content/200-orm/200-prisma-client/600-observability-and-logging/250-opentelemetry-tracing.mdx

```diff
@@ -36,7 +34,7 @@ For each trace, Prisma Client outputs a series of spans. The number and type of
 - `prisma:client:operation`: Represents the entire Prisma Client operation, from Prisma Client to the database and back. It contains details such as the model and method called by Prisma Client. Depending on the Prisma operation, it contains one or more of the following spans:
   - `prisma:client:connect`: Represents how long it takes for Prisma Client to connect to the database.
   - `prisma:client:serialize`: Represents how long it takes to validate and transform a Prisma Client operation into a query for the [query engine](/orm/more/under-the-hood/engines).
-  - `prisma:engine`: Represents how long a query takes in the query engine.
+  - `prisma:engine:query`: Represents how long a query takes in the query engine.
     - `prisma:engine:connection`: Represents how long it takes for Prisma Client to get a database connection.
     - `prisma:engine:db_query`: Represents the database query that was executed against the database. It includes the query in the tags, and how long the query took to run.
     - `prisma:engine:serialize`: Represents how long it takes to transform a database query result into a Prisma Client result.
```

Comment:
We also have separate `prisma:engine:response_json_serialization` since Prisma 5.5.0 (for library engine only, not Accelerate) because this span was confusing to the users since it doesn't include serialization of data between the engine and the client in a sense they expect, it refers to massaging the raw data returned from the database into the shape and types expected by the client.

Reply:
Thanks @aqrln! How does `prisma:engine:response_json_serialization` differ from `prisma:engine:serialize`? If `response_json_serialization` is serialization between engine and client, then is `prisma:engine:serialize` serialization that occurs on the client side?

Reply:
No, `prisma:engine:serialize` happens before `prisma:engine:response_json_serialization` in the engine. What happens there is the engine goes through the raw data returned by the database connector and converts the types according to the schema, fixes up the shapes of the objects (among other things, to compensate for different internal and user-facing representation of virtual selections, like relation count) and so on.